.github/actions/setup-rocm/action.yml

```diff
@@ -87,4 +73,6 @@ runs:
       shell: bash
       run: |
         # All GPUs are visible to the runner; visibility, if needed, will be set by run_test.py.
-        echo "GPU_FLAG=--device=/dev/mem --device=/dev/kfd --device=/dev/dri --group-add video --group-add daemon" >> "${GITHUB_ENV}"
+        # Add render group for container creation.
+        render_gid=`cat /etc/group | grep render | cut -d: -f3`
+        echo "GPU_FLAG=--privileged --device=/dev/mem:/dev/mem:rw --volume /dev/mem:/dev/mem:rw --device=/dev/kfd:/dev/kfd:rw --volume /dev/kfd:/dev/kfd:rw --device /dev/dri:/dev/dri:rw --volume /dev/dri:/dev/dri:rw --group-add video --group-add $render_gid --group-add daemon --security-opt seccomp=unconfined --network=host" >> "${GITHUB_ENV}"
```

Comment:
@saienduri It looks like this snippet includes all the flag settings for the existing nodes too, so it should work for both MI300 and existing nodes? One way of testing if this PR works for existing nodes is by updating one of the yml files to use the non-MI300 label and see if the workflow runs okay.
Cc @amdfaa

Reply:
Yes, it should work for all

Reply:
Left rocm.yml with the old runners and fully rebased so we can see expected behavior